src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketAwaitableEventArgs.cs

```diff
@@ -18,7 +18,7 @@ internal class SocketAwaitableEventArgs : SocketAsyncEventArgs, IValueTaskSource
 
     private readonly PipeScheduler _ioScheduler;
 
-    private Action<object?>? _continuation;
+    private volatile Action<object?>? _continuation;
```

Comment:
I'm guessing this can't be unit-tested.

It's important to add a comment here explaining why `volatile` is required. You can link to issues for more detail/example of the problem.

Reply:
Addressed.

Reply:
Sorry, I'm not sure I follow.  What's the other state that's supposed to be consistent?  It looks like everything else on the object is readonly.

Reply:
Oh, maybe it's `UserToken` from the base type?

Reply:
Yep.